lnwallet/interface.go

```diff
@@ -69,6 +69,12 @@ var (
 	// ErrNotMine is an error denoting that a WalletController instance is
 	// unable to spend a specified output.
 	ErrNotMine = errors.New("the passed output doesn't belong to the wallet")
+
+	// ErrMempoolFee is returned from PublishTransaction in case the
+	// tx being published is not accepted into mempool because the fee
+	// requirements of the mempool backend are not met.
+	ErrMempoolFee = errors.New("transaction rejected: min mempool fee " +
```

Comment:
So let me know what you think about my wrapping here, I did not want to create a specific error type as in the wallet package so I just wrapped it with the `%w` directive. Otherwise it would lead to an inelegant nested structure imo ?

Reply:
Good idea I have done it but I have a question regarding this:

So I allowed to wrap an error message into the lnwallet error. Because this error is now a struct type I use `errors.As` to check the type in other places. I wondered that it is really important to not return a pointer when mapping the errror otherwise type assertions will not match. This feels a bit vulnerable when somebody uses this error and thinks type assertions will match hmm ?

Reply:
I think an easier way to wrap errors in go is to define it as an `errors.New` then return the error using `fmt.Errorf("%w: ...") go give rich content as suggested above, which also allows us to compare errors easily(so no more mix of `Is` and `As`), maybe [this](https://go.dev/blog/go1.13-errors) could help🤓

Reply:
Thank you, did not see the forest in front of the tree 😅

Reply:
I did only wrap the first error and attached the info of the original error from the btcwallet lib, hope that's also what you envisioned ?